packages/react/src/template.tsx

```diff
@@ -174,6 +206,7 @@ export function BaseTemplate({
     propKey: string,
     propValue: DataState,
     parentNode: ComponentDataState | ReferencedComponentDataState,
+    localState: LocalState,
```

Comment:
We need to pass this down from component that calls `renderProp()`, unless we have `use()` hook from canary that allows to access context from helper functions / loops.

Reply:
use will be very handy!

also, have we considered using a different context for each state, so that only the relevant components update when a state changes?